src/normalize-model-and-view.js

```diff
@@ -88,14 +88,15 @@ export function addBunsenModelProperty (bunsenModel, propertyModel, modelPath) {
 
     if (key in pointer) {
       pointer[key] = Object.assign({}, pointer[key])
-    } else if (segments.length === 0) {
-      pointer[key] = propertyModel
-    } else {
+    } else if (segments.length !== 0) {
       pointer[key] = {
         properties: {},
         type: 'object'
       }
     }
+    if (segments.length === 0) {
```

Comment:
I'm guessing this is where the magic is to not just add a new property model but extend an existing one.

Reply:
Actually this is where once we know we have a model path to add, we loop through the original model to add the new model. The problem here was that when the model was already existing, it was going into line 90 but was never going into line 92 even if `segments.length` was equal to zero.

On line 220-230 was where the extend was never produced when a model(object) had children at the same level as the model. In other words we were going inside 221 but we needed to go through the children array also but we never did.